langserve/server.py

```diff
@@ -292,6 +296,23 @@ def _with_validation_error_translation() -> Generator[None, None, None]:
         raise RequestValidationError(e.errors(), body=e.model)
 
 
+def _get_base_run_id_as_str(
+    event_aggregator: AsyncEventAggregatorCallback,
+) -> Optional[str]:
+    """
+    Uses `event_aggregator` to determine the base run ID for a given run. Returns
+    the run_id as a string, or None if it does not exist.
+    """
+    # The first run in the callback_events list corresponds to the
+    # overall trace for request
+    if event_aggregator.callback_events and event_aggregator.callback_events[0].get(
+        "run_id"
+    ):
+        return str(event_aggregator.callback_events[0].get("run_id"))
+    else:
+        None
```

Comment:
Any thoughts on the error handling we'd want here? I just returned none for now, but that's obviously not ideal. Curious what we prefer

Reply:
This should never occur I think so an assertion error is reasonable, can also use log an error once